src/components/DescriptiveRadioButton/DescriptiveRadioButton.tsx

```diff
@@ -0,0 +1,78 @@
+import { ComponentProps, forwardRef } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+import Text from '../Text/Text';
+import { TextVariants } from '../Text';
+
+export interface DescriptiveRadioButtonProps extends Omit<ComponentProps<'input'>, 'type'> {
+  /**
+   * Label for the radio button
+   */
+  label: string;
```

Comment:
@nitpick: `labelText` would align with other implementations and differentiate between a string or an actual label component. Not a blocker or must change